app/instance-initializers/ember-intl.js

```diff
@@ -10,6 +10,7 @@ import { filterBy } from 'ember-intl/utils/initialize';
 export function instanceInitializer(instance) {
     let service = instance.container.lookup('service:intl');
 
+    service.set('appInstance', instance);
```

Comment:
Aye. Felt dirty. Just not sure where else to get the reference for it. I think there is `this.container._registry` but being marked with an underscore and all I didn't investigate further.